packages/@aws-cdk/aws-codepipeline/lib/stage.ts

```diff
@@ -5,6 +5,52 @@ import { cloudformation } from './codepipeline.generated';
 import { Pipeline } from './pipeline';
 import validation = require('./validation');
 
+/**
+ * The construction properties for {@link Stage}.
+ */
+export interface StageProps {
+    /**
+     * Allows specifying where should the newly created {@link Stage}
+     * be placed in the Pipeline.
+     *
+     * @default the stage is added to the end of the Pipeline
+     */
+    readonly placed?: StagePlacement;
+}
+
+/**
+ * Allows you to control where to place a new Stage when it's added to the Pipeline.
+ * Note that you can provide only one of the below properties -
+ * specifying more than one will result in a validation error.
+ *
+ * @see #rightBeforeStage
+ * @see #justAfterStage
+ * @see #atIndex
+ */
+export interface StagePlacement {
```

Comment:
Not sure I think the parent/child semantics make total sense here. Why not just “before” and “after”? What am I missing?

Reply:
It's just a matter of wording. I wanted to make it clear that inserting a Stage before/after a given Stage will "displace" any stage that was before/after that Stage there before (see how clunky this explanation is?), and that's why I came up with the child/parent wording. I'm open to suggestions on how to improve those docs, of course.